### PR TITLE
update eval and fix example

### PR DESCRIPTION
**Pull request by n1ck-guo**

### CODE REVIEW

File: auto_round/__main__.py [hunk 1]
```diff
@@ -348,19 +348,15 @@ def get_library_version(library_name):
...
 def eval(args):
-    quantization_config = AutoRoundConfig(backend=args.device)
     device_str = detect_device(args.device)
-    user_model = AutoModelForCausalLM.from_pretrained(
-        args.model,
-        device_map=device_str, quantization_config=quantization_config)
     model_args = f"pretrained={args.model},trust_remote_code={not args.disable_trust_remote_code}"
     if isinstance(args.tasks, str):
         tasks = args.tasks.split(',')
     res = simple_evaluate(
```
Comment (wenhuach21) [inline auto_round/__main__.py:337]: call get_library_version from the utils

File: auto_round/__main__.py [hunk 2]
```diff
@@ -348,19 +348,15 @@ def get_library_version(library_name):
...
         model="hf",
         model_args=model_args,
-        user_model=user_model,
         tasks=tasks,
+        device=device_str,
```
Comment (wenhuach21) [inline auto_round/__main__.py:341]: parser.add_argument('--eval', action='store_true', 
                        help="whether to use eval mode.")    eval only mode

File: auto_round/__main__.py [hunk 3]
```diff
@@ -348,19 +348,15 @@ def get_library_version(library_name):
...
         batch_size=args.eval_bs)
 
     from lm_eval.utils import make_table  # pylint: disable=E0401
```
Comment (wenhuach21) [inline auto_round/__main__.py:344]: parser.add_argument("--disable_eval", action='store_true',
                        help="Whether to do lmeval evaluation .")   hether to do lm-eval evaluation after tuning

File: auto_round/__main__.py [hunk 4]
```diff
@@ -348,19 +348,15 @@ def get_library_version(library_name):
...
-    user_model = AutoModelForCausalLM.from_pretrained(
-        args.model,
-        device_map=device_str, quantization_config=quantization_config)
     model_args = f"pretrained={args.model},trust_remote_code={not args.disable_trust_remote_code}"
     if isinstance(args.tasks, str):
         tasks = args.tasks.split(',')
     res = simple_evaluate(
         model="hf",
         model_args=model_args,
-        user_model=user_model,
```
Comment (wenhuach21) [inline auto_round/__main__.py:362]: if "opt" in model_name:
        seqlen = model.config.max_position_embeddings
        model.seqlen = model.config.max_position_embeddings
    else:
        seqlen = 2048
        model.seqlen = seqlen
    seqlen = args.seqlen

remove 'model.seqlen = seqlen'  this setting, this is only for ppl evaluation without using lm-eval  model.seqlen = seqlen

File: auto_round/__main__.py [hunk 5]
```diff
@@ -348,19 +348,15 @@ def get_library_version(library_name):
 
 
 def eval(args):
-    quantization_config = AutoRoundConfig(backend=args.device)
     device_str = detect_device(args.device)
-    user_model = AutoModelForCausalLM.from_pretrained(
-        args.model,
-        device_map=device_str, quantization_config=quantization_config)
     model_args = f"pretrained={args.model},trust_remote_code={not args.disable_trust_remote_code}"
```
Comment (wenhuach21) [inline auto_round/__main__.py:334]: reformat this file, there are many extra lines between codes

File: auto_round/__main__.py [hunk 6]
```diff
@@ -205,18 +202,16 @@ def tune(args):
...
-    from auto_round import (AutoRound,
-                            AutoAdamRound)
+    from auto_round import AutoRound, AutoAdamRound
 
     model = model.eval()
     # align with GPTQ to eval ppl
+    seqlen = args.seqlen
     if "opt" in model_name:
         seqlen = model.config.max_position_embeddings
         model.seqlen = model.config.max_position_embeddings
```
Comment (wenhuach21) [inline auto_round/__main__.py:212]: delete

**Review by wenhuach21 — APPROVED** [review]
